src/test/asciidoctorWebViewConverter.test.ts

```diff
@@ -0,0 +1,92 @@
+const processor = asciidoctor()
+
+class TestWebviewResourceProvider implements WebviewResourceProvider {
+  asWebviewUri (resource: vscode.Uri): vscode.Uri {
+    return vscode.Uri.file(resource.path)
+  }
+
+  asMediaWebViewSrc (...pathSegments: string[]): string {
+    return pathSegments.toString()
+  }
+
+  cspSource = 'aaaa'
+}
+
+class TestAsciidocContributions implements AsciidocContributions {
+  readonly previewResourceRoots: ReadonlyArray<vscode.Uri> = []
+  readonly previewScripts: ReadonlyArray<vscode.Uri> = []
+  readonly previewStyles: ReadonlyArray<vscode.Uri> = []
+}
+
+async function testAsciidoctorWebViewConverter (input, antoraResourceUri, expected, root, filePath) {
+  const file = await vscode.workspace.openTextDocument(vscode.Uri.file(`${root}/${filePath}`))
+  const webviewResourceProvider = new TestWebviewResourceProvider()
+  const cspArbitergetSecurityLevelForResource = 2
+  const cspArbiterShouldDisableSecurityWarnings = false
+  const contributions = new TestAsciidocContributions()
+  const previewConfigurations = new AsciidocPreviewConfigurationManager().loadAndCacheConfiguration(file.uri)
+  const asciidoctorWebViewConverter = new AsciidoctorWebViewConverter(
+    file,
+    webviewResourceProvider,
+    cspArbitergetSecurityLevelForResource,
+    cspArbiterShouldDisableSecurityWarnings,
+    contributions,
+    previewConfigurations,
+    antoraResourceUri,
+    undefined
+  )
+  const html = processor.convert(input, { converter: asciidoctorWebViewConverter })
+  assert.strictEqual(html, expected)
+}
+
+suite('AsciidoctorWebViewConverter', async () => {
+  const root = vscode.workspace.workspaceFolders[0].uri.fsPath
+  // WIP need to find more interesting test cases
+  const testCases = [
+    // images
+    {
+      title: 'Should resolve image src',
+      filePath: 'asciidoctorWebViewConverterTest.adoc',
+      input: 'image::images/web-console.png[Couchbase Web Console login]',
+      antoraResourceUri: undefined,
+      expected: `<div class="imageblock">
+<div class="content">
+<img src="images/web-console.png" alt="Couchbase Web Console login">
+</div>
+</div>`,
+    },
+    {
+      title: 'Should resolve image src with Antora id\'s input and Antora support activated',
+      filePath: 'antora/multiComponents/cli/modules/commands/pages/page1.adoc',
+      input: 'image::2.0@cli:commands:seaswell.png[]',
+      antoraResourceUri: `${root}/antora/multiComponents/cli/modules/commands/images/seaswell.png`,
+      expected: `<div class="imageblock">
+<div class="content">
+<img src="${root}/antora/multiComponents/cli/modules/commands/images/seaswell.png" alt="seaswell">
+</div>
+</div>`,
+    },
+    // links
+    {
+      title: 'Should resolve macro link',
+      filePath: 'asciidoctorWebViewConverterTest.adoc',
+      input: 'link:full.adoc[]',
+      antoraResourceUri: undefined,
+      expected: `<div class="paragraph">
+<p><a href="full.adoc" class="undefined" data-href="full.adoc">full.adoc</a></p>
```

Comment:
It's an existing bug (not related to your changes) but it should be `class="bare"`. I think we should call `node.getRole()` (my guess is that `node.role` is `undefined`)

https://github.com/asciidoctor/asciidoctor-vscode/blob/e454a31d1193d714414a9c399e979a3a217ff188/src/asciidoctorWebViewConverter.ts#L187

Anyway, I will create a dedicated issue.